File: noip2015d2t2.pas
var a,b:ansistring;
    i,ans,n,m,k,z:integer;

procedure dfs(x,o:integer);
var i:integer;
begin
    for i:=o to n do
    if a[i]=b[x] then begin
    if i<>o then inc(z);

    writeln(a[i],' ',i,' ',z);
        if x<m then dfs(x+1,i+1);
        if (x=m)and(z>=k) then inc(ans);
    if i<>o then dec(z);
        end;
    
    
end;

begin
    readln(n,m,k);
    readln(a);
    readln(b);
    dfs(1,1);
    writeln(ans);
    end.

File: noip2015d2t2.cpp
#include<iostream>
using namespace std;
char a[1000],b[200];
int w,ans,t,i,y;

int dfs(int x,o)
{
    for (int i=o;i<=n;i++) if (a[i]=b[x]) {if (x<m) ans++,dfs(x+1,i); else ans++};
}

int main()
{
    cin>>n>>m>>k;
    for (y=1;y<=n;y++) cin>>a[i];
    for (y=1;y<=m;y++) cin>>b[i];
    dfs(1,1);
    cout<<ans;
    return 0;
}